src/update.rs

```diff
@@ -0,0 +1,182 @@
+use std::convert::TryInto;
+use std::convert::TryFrom;
+
+use async_trait::async_trait;
+
+use google_datastore1::schemas::{
+  BeginTransactionRequest, BeginTransactionResponse, CommitRequest, CommitResponse, Entity, Key,
+  Mutation, MutationResult,
+};
+
+use crate::entity::{DatastoreEntity};
+
+use crate::error::{DatastorersError, DatastoreClientError};
+use crate::connection::DatastoreConnection;
+
+#[async_trait]
+pub trait DatastorersUpdatable<E> {
+
+  async fn commit(self, connection: &(impl DatastoreConnection + Send + Sync)) -> Result<E, DatastorersError>;
+            
+  async fn delete(self, connection: &(impl DatastoreConnection + Send + Sync)) -> Result<(), DatastorersError>;
+}
+
+
+#[async_trait]
+impl<T> DatastorersUpdatable<T> for T
```

Comment:
I get this error:

`error[E0632]: cannot provide explicit generic arguments when `impl Trait` is used in argument position
  --> src/update.rs:26:6
   |
26 | impl<T> DatastorersUpdatable<T> for T
   |      ^ explicit generic argument not allowed`

Is it not possible to generically implement a trait if one ore more of its methods take an impl!? or what does he mean?

Reply:
`impl Sometype` is sugar for `T: Sometype`, it just means you can't mix syntax. Remove impl and use bounds instead